iroh/src/magicsock/relay_actor.rs

```diff
@@ -266,20 +283,20 @@ impl ActiveRelay {
 
 pub(super) struct RelayActor {
     msock: Arc<MagicSock>,
+    relay_recv_channel: Arc<RelayRecvChannel>,
     /// relay Url -> connection to the node
-    active_relay: BTreeMap<RelayUrl, (mpsc::Sender<ActiveRelayMessage>, JoinHandle<()>)>,
-    msg_sender: mpsc::Sender<ActorMessage>,
+    connected_relays: BTreeMap<RelayUrl, (mpsc::Sender<ConnectedRelayMessage>, JoinHandle<()>)>,
     ping_tasks: JoinSet<(RelayUrl, bool)>,
     cancel_token: CancellationToken,
 }
 
 impl RelayActor {
-    pub(super) fn new(msock: Arc<MagicSock>, msg_sender: mpsc::Sender<ActorMessage>) -> Self {
+    pub(super) fn new(msock: Arc<MagicSock>, recv_channel: Arc<RelayRecvChannel>) -> Self {
```

Comment:
do we ever use  `RelayRecvChannel` not wrapped in an `Arc` if so just internalize it and make it `Clone`?

Reply:
Since this has 3 fields inside, this would make us have another structure a la `Inner`. I find this clearer. Since this is a private struct I would think it's fine?

Reply:
Am I the only one who likes external `Arc`s?  It's nice and explicit, there's nothing wrong with carrying it around: it takes a little extra space but at the same time communicates exactly what this.

I find all this internal Arcing so bothersome, always have to look at the implementation to understand what is going on.  (And you have to deal with all this `Inner` naming! :stuck_out_tongue: ).

The inner fields still need their individual locks anyway, so it's not like it simplifies anything.

Reply:
Split the sender and receiver and now there's no external arc left. :smile: